src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/spreadsheet/PriceSheet.kt

```diff
@@ -1,94 +1,181 @@
+import kotlin.math.max
 
-abstract class PriceSheet(val sheet: Sheet, private val header: Header) {
+abstract class PriceSheet(
+  val sheet: Sheet,
+  private val header: Header,
+  private val subheading: String? = null,
+  private val dataColumns: List<DataColumn> = listOf()
+) {
 
-  protected val rowIndex: AtomicInteger = AtomicInteger(9)
-  protected fun createRow(rIndex: Int = rowIndex.getAndIncrement()): Row = sheet.createRow(rIndex)
-  protected fun getRow(rIndex: Int = rowIndex.getAndIncrement()): Row = sheet.getRow(rIndex)
+  enum class DataColumn(val label: String, val width: Int = 4500) {
+    BILLABLE_JOURNEY_COUNT("Billable journey count"),
+    CANCELLATION_DATE("Cancellation date"),
+    CANCELLATION_TIME("Cancellation time"),
+    CONTRACTOR_BILLABLE("Contractor billable?"),
+    DROP_OFF("Drop off", 9000),
+    DROP_OFF_DATE("Drop off date"),
+    DROP_OFF_TIME("Drop off time"),
+    LOCATION_TYPE("Location Type"),
+    MOVE_DATE("Move date"),
+    MOVE_ID("Move ID"),
+    MOVE_TYPE("Move type"),
+    MOVE_VOLUME("Move volume"),
+    MOVE_VOLUME_WITHOUT_PRICES("Move volume without prices"),
+    NAME("Name"),
+    NOMIS_AGENCY_ID("NOMIS Agency ID"),
+    NOMIS_PRISON_ID("NOMIS prison ID"),
+    NOTES("Notes", 15000),
+    PERCENTAGE("Percentage"),
+    PICK_UP("Pick up", 11000),
+    PICK_UP_DATE("Pick up date"),
+    PICK_UP_TIME("Pick up time"),
+    PRICE("Price"),
+    TOTAL_JOURNEY_COUNT("Total journey count"),
+    TOTAL_PRICE("Total price"),
+    UNIT_PRICE("Unit price"),
+    VEHICLE_REG("Vehicle reg")
+  }
 
-  protected val formatPound = sheet.workbook.createDataFormat().getFormat("\"£\"#,##0.00_);[Red](\"£\"#,##0.00)")
-  protected val formatPercentage = sheet.workbook.createDataFormat().getFormat(BuiltinFormats.getBuiltinFormat(10))
-  protected val formatDate = sheet.workbook.createDataFormat().getFormat("DD/MM/YYYY")
-  protected val formatMonthYear = sheet.workbook.createDataFormat().getFormat("MMMM YYYY")
+  private val rowIndex: AtomicInteger = AtomicInteger(9)
+  private val formatPound = sheet.workbook.createDataFormat().getFormat("\"£\"#,##0.00_);[Red](\"£\"#,##0.00)")
+  private val formatPercentage = sheet.workbook.createDataFormat().getFormat(BuiltinFormats.getBuiltinFormat(10))
+  private val formatDate = sheet.workbook.createDataFormat().getFormat("DD/MM/YYYY")
+  private val formatMonthYear = sheet.workbook.createDataFormat().getFormat("MMMM YYYY")
+  private val colourDarkBlue = XSSFColor(Hex.decodeHex("1b75bc"), null)
+  private val colourLightBlue = XSSFColor(Hex.decodeHex("c9daf8"), null)
+  private val wrapTextRowHeight: Short = 500
 
-  protected val headerMonthYearStyle: CellStyle = sheet.workbook.createCellStyle().apply {
-    fillForegroundColor = IndexedColors.GREY_25_PERCENT.index
-    fillPattern = FillPatternType.SOLID_FOREGROUND
-    dataFormat = formatMonthYear
-    alignment = HorizontalAlignment.LEFT
+  private val fontWhiteArialBold = sheet.workbook.createFont().apply {
```

Comment:
More styling is now needed to allow for what we have lost from the template spreadsheet.